### PR TITLE
Add PrivacyInfo.xcprivacy

### PR DESCRIPTION
**Pull request by garricn**
Addresses #324.

See [Describing use of required reason API](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api) for more information.

| Required Reason API Used          | Used in File                   |
|-----------------------------------|--------------------------------|
| NSFileModificationDate            | PINCache/Source/PINDiskCache.m |
| NSFileModificationDate            | PINCache/Tests/PINCacheTests.m |
| NSURLContentModificationDateKey   | PINCache/Source/PINDiskCache.m |
| NSURLCreationDateKey              | PINCache/Source/PINDiskCache.m |

### CODE REVIEW

**Review by tinder-kylebendelow — APPROVED** [review]

File: PrivacyInfo.xcprivacy [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <key>NSPrivacyCollectedDataTypes</key>
+    <array/>
+    <key>NSPrivacyAccessedAPITypes</key>
+    <array>
+        <dict>
+            <key>NSPrivacyAccessedAPIType</key>
+            <string>NSPrivacyAccessedAPICategoryFileTimestamp</string>
+            <key>NSPrivacyAccessedAPITypeReasons</key>
+            <array>
+                <string>3B52.1</string>
```
Comment (tinder-owenthomas) [inline PrivacyInfo.xcprivacy:18]: Does the SDK provide guarantees that this reason is correct for all the ways people might use it? This is specifically saying that file timestamps will only ever be accessed on files that the end user has given explicit permission to access.

**Comment by hotngui** [comment]
The PrivacyInfo.xcprivacy file also needs to be added to the Package.swift file as a resource to by copied/processed.

**Comment by garricn** [comment]
> When would you like to merge this PR?

Hi @EFFYLYX. Thank you for replying. It would be great if this PR can merge now. Are you able to merge it?

**Comment by farzadshbfn** [comment]
@tinder-owenthomas any chance of merging this PR soon? we're approaching the deadline and it's good to give teams lead-time on updating their frameworks'.

**Review by jparise — APPROVED** [review]

**Comment by jparise** [comment]
This change looks reasonable to me, so I'll help move things along by merging this. That will also give time for folks to adopt this change and provide feedback on anything that might need to be adjusted.

**Comment by gazadge** [comment]
Hey @jparise,
I added a [new PR](https://github.com/pinterest/PINCache/pull/326) related to this issue for including the `.xcprivacy` file as resource when installing PINCache via Cocoapods.

**Pull request merged**